pkg/daemon/writer.go

```diff
@@ -44,33 +45,49 @@ func NewNodeStateStatusWriter(c snclientset.Interface, n string, f func(), devMo
 
 // Run reads from the writer channel and sets the interface status. It will
 // return if the stop channel is closed. Intended to be run via a goroutine.
-func (writer *NodeStateStatusWriter) Run(stop <-chan struct{}, refresh <-chan Message, syncCh chan<- struct{}, destDir string, runonce bool, platformType utils.PlatformType) {
+func (writer *NodeStateStatusWriter) Run(stop <-chan struct{}, refresh <-chan Message, syncCh chan<- struct{}, destDir string, runonce bool, platformType utils.PlatformType) error {
 	glog.V(0).Infof("Run(): start writer")
 	msg := Message{}
 
-	var err error
+	if runonce {
```

Comment:
What about refactoring this if-body to a different function and getting rid of the `runonce` parameters?